scripts/zones/Balgas_Dais/mobs/Macan_Gadangan.lua

```diff
@@ -0,0 +1,78 @@
+-----------------------------------
+-- Area: Balga's Dais
+--  Mob: Macan Gadangan
+-- BCNM: Wild Wild Whiskers
+-----------------------------------
+local ID = require("scripts/zones/Balgas_Dais/IDs")
+require("scripts/globals/spell_data")
+-----------------------------------
+local entity = {}
+-- ELEM
+entity.onMobSpawn = function(mob)
+    mob:setLocalVar("magicRank", 1)
+    mob:setLocalVar("casts", 0)
+    mob:setLocalVar("control", 0)
+
+    mob:addListener("MAGIC_INTERRUPTED", "MACAN_MAGIC_INTERRUPTED", function(mobArg, target, spell, action)
+        mobArg:queue(0, function(mobArg1)
+            mobArg1:useMobAbility(1336)
+        end)
+    end)
+end
+
+local function magicRankUp(mob)
```

Comment:
For styling and keeping things consistent, lets try and keep local functions at the top of the file.